Tests/SwiftParserTest/StatementTests.swift

```diff
@@ -783,3 +783,768 @@ final class StatementTests: XCTestCase {
     )
   }
 }
+
+final class ThenStatementTests: XCTestCase {
+  // Overload the module-level assertParse with one that enables 'then' statements.
+  func assertParse(
+    _ markedSource: String,
+    substructure expectedSubstructure: Syntax? = nil,
+    substructureAfterMarker: String = "START",
+    diagnostics expectedDiagnostics: [DiagnosticSpec] = [],
+    applyFixIts: [String]? = nil,
+    fixedSource expectedFixedSource: String? = nil,
+    options: AssertParseOptions = [],
+    experimentalFeatures: Parser.ExperimentalFeatures = .thenStatements,
+    file: StaticString = #file,
+    line: UInt = #line
+  ) {
+    SwiftParserTest.assertParse(
+      markedSource,
+      substructure: expectedSubstructure,
+      substructureAfterMarker: substructureAfterMarker,
+      diagnostics: expectedDiagnostics,
+      applyFixIts: applyFixIts,
+      fixedSource: expectedFixedSource,
+      options: options,
+      experimentalFeatures: experimentalFeatures,
+      file: file,
+      line: line
+    )
+  }
```

Comment:
> just pass `.thenStatements` to every `assertParse` call

IMO that seems like it would be quite easy to accidentally forget to do

> add a `class ParserTestCase: XCTestCase`

This is probably my preferred solution if we don't want to overload

Reply:
Updated to introduce a new base class